### ISSUE TITLE
extensions.json is not generated correctly when ReadyToRun is enabled

### ISSUE THREAD

**Issue by shibayan** [Azure/azure-functions-host#6540]
With ReadyToRun enabled, the ExtensionsMetadataGenerator reports an assembly loading error and does not generate the required extensions.json correctly.

NET Core on x64 because it is not possible to load R2R pre-compiled assemblies for win-x86.

>C:\Users\shibayan\.nuget\packages\microsoft.azure.webjobs.script.extensionsmetadatagenerator\1.1.8\build\Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator.targets(37,5): warning :     Could not evaluate 'FunctionApp6.dll' for extension metadata. Exception message: Could not load file or assembly 'FunctionApp6, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. [C:\Users\shibayan\source\repos\FunctionApp6\FunctionApp6\FunctionApp6.csproj

#### Repro steps

Provide the steps required to reproduce the problem:

1. Create new function app
2. Install Durable Functions (v2.2.2)
3. Add new orchestrator function
4. Add ReadyToRun configuration to csproj (https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-class-library#readytorun)

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <AzureFunctionsVersion>v3</AzureFunctionsVersion>
    <PublishReadyToRun>true</PublishReadyToRun>
    <RuntimeIdentifier>win-x86</RuntimeIdentifier>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.DurableTask" Version="2.2.2" />
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.9" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
</Project>
```

5. Exec `dotnet publish -c Release -o ./publish`

#### Expected behavior

The DurableTask extension will be correctly registered in extensions.json.

```json
{
  "extensions":[
    { "name": "DurableTask", "typeName":"Microsoft.Azure.WebJobs.Extensions.DurableTask.DurableTaskWebJobsStartup, Microsoft.Azure.WebJobs.Extensions.DurableTask, Version=2.0.0.0, Culture=neutral, PublicKeyToken=014045d636e89289"}
  ]
}
```

#### Actual behavior

extensions.json is generated as an empty file.

```json
{
  "extensions":[
  ]
}
```

#### Known workarounds

Match .NET Core bitness to `RuntimeIdentifier`. However, in CI (Azure Pipelines / GitHub Actions) environments, it's difficult to solve this problem because I can't select bitness.

**Comment by fabiocav** [comment]
@brettsam can you please share a link to the updated metadata generator once it is published as it will hopefully address this as well?

**Comment by brettsam** [comment]
Yes, #6052 should fix this.

Please try adding a direct reference to this package (version 1.2.0) in your csproj: https://www.nuget.org/packages/Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator/1.2.0

Soon this dependency will be updated in Microsoft.Net.Sdk.Functions, but for now you have to directly reference this newest version.

Closing this issue as it should be resolved, but please re-activate if you're still able to repro.

**Comment by shibayan** [comment]
Thanks, I've confirmed that the problem is resolved in ExtensionMetadataGenerator v1.2.0.
